### PR TITLE
#272 Login change error -DONE

### LINKED ISSUE #272: [7] Login change error
Once the Client tries to change the login (Settings > Account > Personal data) to already existing then the error occurs. There's no information about the limitations like it's on the Clients tab (Worker/Accountant/Admin roles) once the user tries to create a new user but the provided login already exists.

Place where the error occurs:
image

```
2020-12-23 14:32:30 ERROR org.hibernate.engine.jdbc.spi.SqlExceptionHelper     - ERROR: duplicate key value violates unique constraint "ukm2tkxarwnquqaiuqnjmyqkaui"
  Szczegóły: Key (login)=(TestAAP) already exists.
Exception in thread "AWT-EventQueue-0" javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:181)
	at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:188)
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1356)
	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:443)
	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:3202)
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:2370)
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:447)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:183)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.access$300(JdbcResourceLocalTransactionCoordinatorImpl.java:40)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:281)
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101)
	at com.javafee.elibrary.hibernate.dao.HibernateUtil.commitTransaction(HibernateUtil.java:84)
	at com.javafee.elibrary.core.settingsform.PersonalDataChangePanelEvent.onClickBtnAccept(PersonalDataChangePanelEvent.java:91)
	at com.javafee.elibrary.core.settingsform.PersonalDataChangePanelEvent.lambda$control$0(PersonalDataChangePanelEvent.java:45)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
```

Validation message which should be displayed:
image

### PR DESCRIPTION
**Pull request by MichalJagodzinski**

### CODE REVIEW

File: elibrary-core/source/src/controller/java/com/javafee/elibrary/core/settingsform/PasswordChangePanelEvent.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package com.javafee.elibrary.core.settingsform;
 
 import java.text.MessageFormat;
+import java.util.Objects;
```
Comment (JaRutkowski) [inline elibrary-core/source/src/controller/java/com/javafee/elibrary/core/settingsform/PasswordChangePanelEvent.java]: Are those two imports the only change in the file? 
If yes, then please remove it. 🗑️

File: elibrary-core/source/src/controller/java/com/javafee/elibrary/core/startform/RegistrationEvent.java [hunk 2]
```diff
@@ -49,29 +49,16 @@ public static void forceClearRegistrationEvenet() {
...
-		return result;
+		return performCheck(userAccount, password);
 	}
 
 	private static boolean performCheck(UserAccount userAccount, String password) {
 		boolean result = false;
 		if (userAccount != null)
 			Params.getInstance().add("ALREADY_REGISTERED", RegistrationFailureCause.ALREADY_REGISTERED);
-		else if (Common.checkPasswordStrength(password))
+		if (Common.checkPasswordStrength(password))
```
Comment (JaRutkowski) [inline elibrary-core/source/src/controller/java/com/javafee/elibrary/core/startform/RegistrationEvent.java:61]: Could u please make it shorter? 🚀 Please check if the ternary operator could be used here to simplify the code.

**Review by JaRutkowski — CHANGES_REQUESTED** [review]
Please see the comments. 💬 
Once the fixes will be delivered 🔨 📦  please answer the comment and resolve the conversation. ✅ 
If u think that the requested change is not applicable/not required then don't hesitate to write it providing some arguments. 💪

File: elibrary-core/source/src/controller/java/com/javafee/elibrary/core/settingsform/PasswordChangePanelEvent.java (hunk 1 above)
Comment (MichalJagodzinski) [reply elibrary-core/source/src/controller/java/com/javafee/elibrary/core/settingsform/PasswordChangePanelEvent.java]: This is done

File: elibrary-core/source/src/controller/java/com/javafee/elibrary/core/startform/RegistrationEvent.java (hunk 2 above)
Comment (MichalJagodzinski) [reply elibrary-core/source/src/controller/java/com/javafee/elibrary/core/startform/RegistrationEvent.java:61]: I tried but there is a problem with method add(), it is void and if we want use here trenary operator this should returns boolean, so i think this change is not effective.

**Review by JaRutkowski — APPROVED** [review]

**Pull request merged**